packages/snjs/lib/models/app/search.ts

```diff
@@ -0,0 +1,40 @@
+/** Grouping of required state variables to perform the search */
+
+import { CollectionSort } from "@Lib/protocol/collection/item_collection";
+import { SNTag } from "./tag";
+
+export interface SearchPayload {
+  tag: SNTag,
+  sortBy: CollectionSort,
+  isReversedSort: "asc" | "dsc" | undefined,
+  showArchiveOrTrashed: boolean,
+  hidePinned: boolean,
+  query: string
+
+}
+
+export class Search {
+
+  /**
+   * This is a static class so instantiation should be impossible.
+   */
+  private constructor() {
+  }
+
+  public static create(tag: SNTag,
```

Comment:
Is a utility function to create the `SearchPayload` needed here, since it does nothing special? From a client-side perspective,

```ts
application.setNotesDisplayOptions({
  sortyBy,
  // ...
});
```

Seems nicer than:

```ts
application.setNotesDisplayOptions(Search.create(/* ... */));
```

Especially because we get named arguments.

Reply:
No, I was creating a class in the event that SearchPayload became more complex. I should have followed YAGNI.